test/mobile/custom_build/build.sh

```diff
@@ -0,0 +1,145 @@
+#!/bin/bash
+###############################################################################
+# This script shows/tests the flow to build libtorch locally with optimized
+# binary size for mobile devices and the flow to integrate it with a simple
+# predictor in c++.
+#
+# There are three custom build types:
+#
+# 1. `TEST_DEFAULT_BUILD=1 ./build.sh` - it is similar to the prebuilt libtorch
+# libraries released for Android and iOS (same CMake build options + host
+# toolchain), which doesn't contain autograd function nor backward ops thus is
+# smaller than full LibTorch.
+#
+# 2. `TEST_CUSTOM_BUILD_STATIC=1 ./build.sh` - it further optimizes libtorch
+# size by only including ops used by a specific model.
+#
+# 3. `TEST_CUSTOM_BUILD_DYNAMIC=1 ./build.sh` - similar as 2) except that it
+# relies on the op dependency graph (instead of static dispatch) to calculate
+# and keep all transitively dependent ops by the model.
+# Note that LLVM_DIR environment variable should be set to the location of
+# LLVM-dev toolchain.
+#
+# Type 2) will be deprecated by type 3) in the future.
+###############################################################################
+
+set -ex -o pipefail
+
+SRC_ROOT="$( cd "$(dirname "$0")"/../../.. ; pwd -P)"
+TEST_SRC_ROOT="${SRC_ROOT}/test/mobile/custom_build"
+BUILD_ROOT="${BUILD_ROOT:-${SRC_ROOT}/build_test_custom_build}"
+mkdir -p "${BUILD_ROOT}"
+cd "${BUILD_ROOT}"
+
+prepare_model_and_dump_root_ops() {
+  cd "${BUILD_ROOT}"
+  MODEL="${BUILD_ROOT}/MobileNetV2.pt"
+  ROOT_OPS="${BUILD_ROOT}/MobileNetV2.yaml"
+
+  if [ ! -f "${MODEL}" ] || [ ! -f "${ROOT_OPS}" ]; then
+    python "${TEST_SRC_ROOT}/prepare_model.py"
```

Comment:
If this script is intended to be run locally by developers who might run this repeatedly, it might be a good idea to unconditionally rerun `prepare_model.py` if its mtime is greater than MODEL/ROOT_OPS

Reply:
Kind of wondering if this should have been written in make/cmake :)